File: 021 Longest Peak/LongestPeak.js
function longestPeak(array){
	let longestPeakLength = 0;
	let i = 1;
	while (i < array.length - 1) {
		let isPeak = array[i] > array[i - 1] && array[i] > array[i + 1];
		if(!isPeak){
			i++;
			continue
		}
		let leftIndex = i - 2;
		while(leftIndex >= 0 && array[leftIndex] < array[leftIndex + 1] ){
			leftIndex--;
		}
		let rightIndex = i + 2;
		while(rightIndex < array.length && array[rightIndex] < array[rightIndex - 1] ){
			rightIndex++;
		}
		const currentLength = rightIndex - leftIndex - 1;
		longestPeakLength = Math.max(longestPeakLength, currentLength);
		i = rightIndex
	}
	return longestPeakLength;
}

File: 021 Longest Peak/LongestPeak.py
def longestPeak(array):
    i = 1
    longestPeakLength = 0
    while i < len(array) - 1:
        isPeak = array[i-1] < array[i] and array[i] > array[i + 1]
        if not isPeak:
            i += 1
            continue
        # check how long the peak goes from left indexes and right indexes
        
        leftIndex = i - 2
        while leftIndex >= 0 and array[leftIndex] < array[leftIndex + 1]:
            leftIndex -= 1

        rightIndex = i + 2
        while rightIndex < len(array) and array[rightIndex] < array[rightIndex - 1]:
            rightIndex += 1

        currentPeakLength = rightIndex - leftIndex - 1
        longestPeakLength = max(longestPeakLength, currentPeakLength)
        i = rightIndex
    return longestPeakLength
